### ISSUE TITLE
Editing Navigation Menus via Site Editor is a UX regression.

### ISSUE THREAD

**Issue by ghost** [WordPress/gutenberg#53881]
## What problem does this address?

Prior to the Site Editor being introduced, it was possible to get to the Menus admin page (`/wp-admin/nav-menus.php`) in a single click:

- From Dashboard: Hover over "Appearance", click "Menus".
- From frontend: Hover over site name in Admin Bar, click "Menus".

With the introduction of the expanded Site Editor in 6.3, this navigation flow is now much more complicated:

- From Dashboard (2 clicks): 
  - Hover over "Appearance", click "Editor"
  - Click on "Navigation"
- From frontend (4 clicks):
  - Click on "Edit Site" in the Admin Bar
  - You're now looking at a Page Template (??) instead of the Site Editor "Design" menu
  - Click "<" icon to go back to "Templates"
  - Click "<" icon again to go back to "Design"
  - Click on "Navigation"

Further, from the "Navigation" menu, if you want to *add* any new items, you have to then click on the little pencil icon (with no label) in order to load the Navigation block and then edit it from there.

![wp-6-3-nav-menu-main](https://github.com/WordPress/gutenberg/assets/10351660/582e064b-8931-4449-8795-b6777565c256)

It is also worth noting that removing/reordering items via the Navigation *menu* and via the Navigation *block* are completely different editing workflows.

## What is your proposed solution?

- Clicking on "Appearance > Editor" and "Admin Bar > Edit Site" should go to the same place: `/wp-admin/site-editor.php` - Clicking on Edit Site from the Admin Bar should NOT take you to a Page Template.

### If there is only one menu:

- The "Navigation" menu (`/wp-admin/site-editor.php?path=%2Fnavigation`) is skipped entirely
- Clicking on "Navigation" from the Site Editor "Design" menu should go directly to the Navigation *block* editor. (`/wp-admin/site-editor.php?postId=X&postType=wp_navigation&canvas=edit`)

### If there are multiple menus:

- The "Navigation" menu is shown with a list of all menus (as it currently is) ![wp-6-3-nav-menu-multiple](https://github.com/WordPress/gutenberg/assets/10351660/ad8aa757-f5b0-4f24-8c91-c857d61db7ba) 
(also why is this not alphabetical?)

- Clicking on each particular menu in the "Navigation" menu should go directly to the Navigation *block* editor. (`/wp-admin/site-editor.php?postId=X&postType=wp_navigation&canvas=edit`) 
![wp-6-3-nav-block-editor](https://github.com/WordPress/gutenberg/assets/10351660/13c9bed1-8ca4-483a-a12d-d630faa64b3a)

## TL;DR

In essence, the screen `/wp-admin/site-editor.php?postId=X&postType=wp_navigation` is pretty redundant and should be removed.

This would still be more clicks than the prior-to-6.3 system, but it would be a more consistent and less confusing editing flow at least.

And opening the Site Editor from the Dashboard or the frontend should go to the same place.

**Comment by Mayank-Tripathi32** [comment]
I think we can close this issue, as its already been implemented. 

> Clicking on "Navigation" from the Site Editor "Design" menu should go directly to the Navigation block editor. (/wp-admin/site-editor.php?postId=X&postType=wp_navigation&canvas=edit)

Not sure, if this was discussed but currently it still opens navigation.

**Comment by jasmussen** [comment]
Per the conversation on this thread, and a little digging, the core of this issue appears to be resolved. The specific pain point described, having to click through a list screen and an unlabeled pencil icon to reach the Navigation block editor, was addressed by #51565 ("Remove custom button and (conditionally) show single menu on Navigation route in Browse Mode"). In current trunk, when a site has a single navigation menu, SidebarNavigationScreenNavigationMenus skips the list screen and renders the menu's block-editing UI directly (see packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/single-navigation-menu.js and navigation-menu-editor.js), matching the "direct to block editor" behavior requested here. 

In that vein, I'll close this one as fixed, simply to consolidate efforts on other open issues.

But I also want to acknowledge the subtext: that navigation editing is at present not an ideal experience. The other open issues with the same Navigation Menus label cover this as well, and is what I would direct to. Specifically this one is a good one to put a spotlight on: #76037
